task/monorail-server/src/bsp/sidecar_bcd.rs

```diff
@@ -472,6 +500,15 @@ impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
             }
         }
 
+        if let VLanMode::UnlockedUntil(t) = self.vlan_mode {
+            let now = userlib::sys_get_timer().now;
```

Comment:
If the state (t) is lost due to task restart, are the VLANs locked on task restart?

Reply:
Yes, when the `monorail` task initially starts up, the switch is configured to be locked.

Note that if just one of the lock-sensitive tasks (`control-plane-agent`, `monorail`, `net`) reboots, it may be locked while the rest of the system is unlocked.  I think this is fine, since it's failing into the more secure state.

Reply:
> Note that if just one of the lock-sensitive tasks (`control-plane-agent`, `monorail`, `net`) reboots, it may be locked while the rest of the system is unlocked. I think this is fine, since it's failing into the more secure state.

I think this is minor, but could we land in a "system is unlocked but we can't lock it" or vice versa because of state disagreements? E.g., if CPA reboots and now thinks we're locked, will it refuse requests to lock even though it should pass them on?

Since all of this is time based it would clear itself up after at most an hour, and for that hour I agree we'd have failed in the correct direction.

Reply:
If CPA reboots and thinks we're locked, it will ignore lock calls (because they're not on the list of messages accepted while locked), but _will_ accept calls to unlock (which will just unlock itself and update the unlock timer for other tasks).

My intent is for an unlock command to always unlock the system, even if some parts were already unlocked; let me know if you spot any places where that seems incorrect.